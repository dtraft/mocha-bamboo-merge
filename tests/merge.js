const { merge } = require('../lib')
const path = require('path')

describe('merge', () => {
  test('merges configs', async () => {
    const report = await merge({
      reportDir: path.resolve(__dirname, './mocha-bamboo-report'),
    })

    expect(report.stats).toMatchSnapshot({
      start: expect.any(String),
      end: expect.any(String),
    })

    expect(report.failures.length).toBe(3)
    expect(report.passes.length).toBe(3)
    expect(report.skipped.length).toBe(0)
  })

  test('throws when invalid directory provided', async () => {
    const dirname = './invalid-directory'
    const dir = path.resolve(process.cwd(), dirname)
    await expect(merge({ reportDir: dirname })).rejects.toEqual(
      new Error(`Directory ${dir} does not exist`)
    )
  })

  test('defaults to mocha-bamboo-report directory', async () => {
    const dir = path.resolve(process.cwd(), 'mocha-bamboo-report')
    await expect(merge()).rejects.toEqual(
      new Error(`Directory ${dir} does not exist`)
    )
  })
})
