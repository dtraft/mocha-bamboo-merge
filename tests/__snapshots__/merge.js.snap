// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`merge merges configs 1`] = `
Object {
  "duration": NaN,
  "end": Any<String>,
  "failures": 3,
  "passes": 3,
  "pending": 0,
  "start": Any<String>,
  "suites": 3,
  "tests": 6,
}
`;
